prow/kube/prowjob.go

```diff
@@ -48,11 +48,13 @@ const (
 )
 
 type ProwJob struct {
-	APIVersion string        `json:"apiVersion,omitempty"`
-	Kind       string        `json:"kind,omitempty"`
-	Metadata   ObjectMeta    `json:"metadata,omitempty"`
-	Spec       ProwJobSpec   `json:"spec,omitempty"`
-	Status     ProwJobStatus `json:"status,omitempty"`
+	APIVersion   string        `json:"apiVersion,omitempty"`
+	Kind         string        `json:"kind,omitempty"`
+	Metadata     ObjectMeta    `json:"metadata,omitempty"`
+	Spec         ProwJobSpec   `json:"spec,omitempty"`
+	Status       ProwJobStatus `json:"status,omitempty"`
+	ParentStatus ProwJobStatus `json:"parent_status,omitempty"`
+	ParentJob    string        `json:"parent_job,omitempty"`
```

Comment:
https://github.com/kubernetes/community/blob/548eb7292d45a4af78e0141d33094130dfe6080a/contributors/design-proposals/garbage-collection.md
https://github.com/kubernetes/community/blob/31d62c70214d66ca2c21716c02b83d9087fdf3ff/contributors/design-proposals/controller-ref.md

It doesn't make sense to fully support owner references (adoption, orphaning) but we could use them for denoting relationships between prowjobs.